x/profiles/keeper/msg_server_chain_link.go

```diff
@@ -0,0 +1,117 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/desmos-labs/desmos/x/profiles/types"
+)
+
+func (k msgServer) LinkChainAccount(goCtx context.Context, msg *types.MsgLinkChainAccount) (*types.LinkChainAccountResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	if err := msg.ValidateBasic(); err != nil {
+		return nil, err
+	}
+
+	if err := msg.SourceProof.Verify(k.cdc); err != nil {
+		return nil, err
+	}
+
+	if err := msg.DestinationProof.Verify(k.cdc); err != nil {
+		return nil, err
+	}
+
+	// Check if address has the profile
+	profile, found, err := k.GetProfile(ctx, msg.DestinationAddress)
+	if err != nil {
+		return nil, err
+	}
+	if !found {
+		return nil, fmt.Errorf("address does not have any profile")
+	}
+
+	link := types.NewChainLink(
+		msg.SourceAddress,
+		msg.SourceProof,
+		msg.SourceChainConfig,
+		ctx.BlockTime(),
+	)
+
+	if err := k.StoreChainLink(ctx, link); err != nil {
+		return nil, err
+	}
+
+	// Store chain link to the profile
+	profile.ChainsLinks = append(profile.ChainsLinks, link)
+	if err := k.StoreProfile(ctx, profile); err != nil {
+		k.DeleteChainLink(ctx, link.ChainConfig.Name, link.Address)
+		return nil, err
+	}
```

Comment:
This should be included inside the `StoreChainLink` to avoid having to remember that they should be called along with it.